### PR TITLE
pythonPackages.rfc3986: fix pytest dependency

### PR DESCRIPTION
**Pull request by sveitser**
###### Motivation for this change
Build fails with
```
/nix/store/63karsgdg7fm3q0if4zfd7apbd8ac1ci-stdenv-linux/setup: line 1311: py.test: command not found
```

https://hydra.nixos.org/build/89616980/nixlog/1/tail

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/rfc3986/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@ buildPythonPackage rec {
     sha256 = "1qf4dyxvjs7mxrxc0gr7gzyn4iflb2wgq01r5pzrxac8rnvy8fmw";
   };
 
-  buildInputs = [ pytest ];
+  checkInputs = [ pytest ];
   checkPhase = ''
     py.test
```
Comment (worldofpeace) [inline pkgs/development/python-modules/rfc3986/default.nix]:
```suggestion
    pytest
```

Old command

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build pythonPackages.rfc3986 python3Packages.rfc3986

File: pkgs/development/python-modules/rfc3986/default.nix [hunk 2]
```diff
@@ -10,9 +10,9 @@ buildPythonPackage rec {
     sha256 = "1qf4dyxvjs7mxrxc0gr7gzyn4iflb2wgq01r5pzrxac8rnvy8fmw";
   };
 
-  buildInputs = [ pytest ];
+  checkInputs = [ pytest ];
   checkPhase = ''
-    py.test
```
Comment (FRidh) [inline pkgs/development/python-modules/rfc3986/default.nix]: unnecessary change

Comment (FRidh) [reply pkgs/development/python-modules/rfc3986/default.nix]: ah, nevermind, I did not know `pytest` was the preferred invocation.

Comment (worldofpeace) [reply pkgs/development/python-modules/rfc3986/default.nix]: Indeed, but might as well change it right?

**Comment by worldofpeace** [comment]
:rofl:  That should have been a squash

**Comment by worldofpeace** [comment]
Backported in https://github.com/NixOS/nixpkgs/commit/1b97770dcf756b1cac048f8af8567747338561b7
